apollo-router/src/plugins/record_replay/replay.rs

```diff
@@ -0,0 +1,418 @@
+                                    print!("{}", style(change).green().underlined());
+                                } else {
+                                    print!("{}", style(change).green());
+                                }
+                            }
+                        }
+                        similar::ChangeTag::Delete => {
+                            has_changes = true;
+                            print!(
+                                "{:>5} {:>5} │{}",
+                                style(change.old_index().unwrap()).cyan().dim(),
+                                "",
+                                style("-").red(),
+                            );
+                            for &(emphasized, change) in change.values() {
+                                let change = render_invisible(change, newlines_matter);
+                                if emphasized {
+                                    print!("{}", style(change).red().underlined());
+                                } else {
+                                    print!("{}", style(change).red());
+                                }
+                            }
+                        }
+                        similar::ChangeTag::Equal => {
+                            print!(
+                                "{:>5} {:>5} │ ",
+                                style(change.old_index().unwrap()).cyan().dim(),
+                                style(change.new_index().unwrap()).cyan().dim().bold(),
+                            );
+                            for &(_, change) in change.values() {
+                                let change = render_invisible(change, newlines_matter);
+                                print!("{}", style(change).dim());
+                            }
+                        }
+                    }
+                    if change.missing_newline() {
+                        println!();
+                    }
+                }
+            }
+        }
+
+        if !has_changes {
+            println!(
+                "{:>5} {:>5} │{}",
+                "",
+                style("-").dim(),
+                style(" snapshots are matching").cyan(),
+            );
+        }
+
+        println!("────────────┴{:─^1$}", "", width.saturating_sub(13));
+    }
+}
+
+fn print_line(width: usize) {
+    println!("{:═^1$}", "", width);
+}
+
+fn render_invisible(s: &str, newlines_matter: bool) -> Cow<'_, str> {
+    if newlines_matter || s.find(&['\x1b', '\x07', '\x08', '\x7f'][..]).is_some() {
+        Cow::Owned(
+            s.replace('\r', "␍\r")
+                .replace('\n', "␊\n")
+                .replace("␍\r␊\n", "␍␊\r\n")
+                .replace('\x07', "␇")
+                .replace('\x08', "␈")
+                .replace('\x1b', "␛")
+                .replace('\x7f', "␡"),
+        )
+    } else {
+        Cow::Borrowed(s)
+    }
+}
+
+#[cfg(test)]
+#[path = "replay_tests.rs"]
+mod tests;
```

Comment:
I think it's more idiomatic to get rid of that part and start the `replay_tests.rs` file with a global attribute like this 
```rust
#![cfg(test)]
// Content of the test file
```

Reply:
i moved the cfg(test) to the test file